src/selectors/assets.js

```diff
@@ -18,134 +18,144 @@
     51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 */
 
-import get from 'lodash.get';
 import { createSelector } from 'reselect';
+import { mapValues } from 'lodash';
 
 // utils
-import {
-  findFirstArchanovaAccount,
-  findFirstEtherspotAccount,
-  getAccountId,
-  getEnabledAssets,
-} from 'utils/accounts';
+import { findFirstArchanovaAccount, getAccountId } from 'utils/accounts';
 import {
   getAssetData,
   getAssetsAsList,
   getBalanceBN,
   getBalanceInFiat,
   getFormattedBalanceInFiat,
+  mapWalletAssetsBalancesIntoAssets,
 } from 'utils/assets';
 import { reportErrorLog } from 'utils/common';
 
-// constants
-import { DEFAULT_ACCOUNTS_ASSETS_DATA_KEY } from 'constants/assetsConstants';
-
 // types
-import type { Asset, Assets, Rates } from 'models/Asset';
-import type { RootReducerState } from 'reducers/rootReducer';
-import type { CategoryBalancesPerChain } from 'models/Balances';
+import type {
+  SupportedAssetsPerChain,
+  Rates,
+  Assets,
+  Asset,
+} from 'models/Asset';
+import type {
+  AssetBalancesPerAccount,
+  CategoryBalancesPerChain,
+  WalletAssetsBalances,
+} from 'models/Balances';
+import type { ChainRecord } from 'models/Chain';
+import type { Account } from 'models/Account';
 
-import { accountAssetsBalancesSelector } from 'selectors/balances';
 import {
-  assetsSelector,
+  accountAssetsBalancesSelector,
+  accountEthereumWalletAssetsBalancesSelector,
+} from 'selectors/balances';
+import {
   activeAccountIdSelector,
-  hiddenAssetsSelector,
   supportedAssetsSelector,
   accountsSelector,
   ratesSelector,
   baseFiatCurrencySelector,
+  assetsBalancesSelector,
 } from './selectors';
 
 
+export const ethereumSupportedAssetsSelector = createSelector(
+  supportedAssetsSelector,
+  (supportedAssets: SupportedAssetsPerChain): Asset[] => supportedAssets?.ethereum ?? [],
+);
```

Comment:
No need to memoize it, as there is no computation
```suggestion
export const ethereumSupportedAssetsSelector = (root: Root...) => root.assets.supportedAssets.ethereum ?? [];

Reply:
I found this easier to reuse/replace in all legacy components.